plugins/inputs/win_eventlog/win_eventlog.go

```diff
@@ -0,0 +1,515 @@
+  '''
+
+  ## System field names:
+  ##   "Source", "EventID", "Version", "Level", "Task", "Opcode", "Keywords", "TimeCreated",
+  ##   "EventRecordID", "ActivityID", "RelatedActivityID", "ProcessID", "ThreadID", "ProcessName",
+  ##   "Channel", "Computer", "UserID", "UserName", "Message", "LevelText", "TaskText", "OpcodeText"
+
+  ## In addition to System, Data fields can be unrolled from additional XML nodes in event.
+  ## Human-readable representation of those nodes is formatted into event Message field,
+  ## but XML is more machine-parsable
+
+  # Process UserData XML to fields, if this node exists in Event XML
+  process_userdata = true
+
+  # Process EventData XML to fields, if this node exists in Event XML
+  process_eventdata = true
+
+  ## Separator character to use for unrolled XML Data field names
+  separator = "_"
+
+  ## Get only first line of Message field. For most events first line is usually more than enough
+  only_first_line_of_message = true
+
+  ## Fields to include as tags. Globbing supported ("Level*" for both "Level" and "LevelText")
+  event_tags = ["Source", "EventID", "Level", "LevelText", "Task", "TaskText", "Opcode", "OpcodeText", "Keywords", "Channel", "Computer"]
+
+  ## Default list of fields to send. All fields are sent by default. Globbing supported
+  event_fields = ["*"]
+
+  ## Fields to exclude. Also applied to data fields. Globbing supported
+  exclude_fields = ["Binary", "Data_Address*"]
+
+  ## Skip those tags or fields if their value is empty or equals to zero. Globbing supported
+  exclude_empty = ["*ActivityID", "UserID"]
+`
+
+// WinEventLog config
+type WinEventLog struct {
+	Locale                 uint32   `toml:"locale"`
+	EventlogName           string   `toml:"eventlog_name"`
+	Query                  string   `toml:"xpath_query"`
+	ProcessUserData        bool     `toml:"process_userdata"`
+	ProcessEventData       bool     `toml:"process_eventdata"`
+	Separator              string   `toml:"separator"`
+	OnlyFirstLineOfMessage bool     `toml:"only_first_line_of_message"`
+	EventTags              []string `toml:"event_tags"`
+	EventFields            []string `toml:"event_fields"`
+	ExcludeFields          []string `toml:"exclude_fields"`
+	ExcludeEmpty           []string `toml:"exclude_empty"`
+	subscription           EvtHandle
+	buf                    []byte
+	Log                    telegraf.Logger
+}
+
+var bufferSize = 1 << 14
+
+var description = "Input plugin to collect Windows Event Log messages"
+
+// Description for win_eventlog
+func (w *WinEventLog) Description() string {
+	return description
+}
+
+// SampleConfig for win_eventlog
+func (w *WinEventLog) SampleConfig() string {
+	return sampleConfig
+}
+
+// Gather Windows Event Log entries
+func (w *WinEventLog) Gather(acc telegraf.Accumulator) error {
+
+	var err error
+	if w.subscription == 0 {
+		w.subscription, err = w.evtSubscribe(w.EventlogName, w.Query)
+		if err != nil {
+			w.Log.Error("Subscription error:", err.Error())
+		}
```

Comment:
Is there any point in continuing when `w.evtSubscribe(w.EventlogName, w.Query)` returns error and thus `w.subscription == 0`?